contracts/RewardPool.sol

```diff
@@ -111,7 +110,7 @@ contract RewardPool is Initializable, Versionable, Ownable, ReentrancyGuard {
         (address, uint256, uint256, uint256, uint256, address, bytes)
       );
     if (block.number >= startBlock && block.number < endBlock) {
```

Comment:
Hmmmm, I might have missed this. how does tally know of both the startBlock and endBlock of a payment cycle?

It could be that enforcing tally to know this information may add to some complexity.

Reply:
Yeah with what I've done tally would need to know the range each payment cycle covers - I'll come back to that in another comment but the code here is actually the range of validity for the claim rather than anything to do with the payment cycle. I'll rename these variables as I think it's confusing now you ask about them.

e.g. You have reward of 500 airmiles but the claim expires in a year and after that point if you've not claimed them that's it.

Start block is "valid from" and end block is "valid to" for the claim.

Reply:
Right goddit. This is the expiry. Cool